### PR TITLE
review dependencies

### PR DESCRIPTION
**Pull request by fmessmer**
sorting, cleaning up and review (used roscompile)

### CODE REVIEW

File: cob_base_drive_chain/CMakeLists.txt [hunk 1]
```diff
@@ -1,42 +1,29 @@
...
 
-###################################
-## catkin specific configuration ##
-###################################
 
 catkin_package(
-    CATKIN_DEPENDS roscpp control_msgs cob_generic_can cob_canopen_motor cob_utilities libpcan std_msgs std_srvs sensor_msgs diagnostic_msgs message_runtime
-    INCLUDE_DIRS common/include
-    LIBRARIES ${PROJECT_NAME}
+    CATKIN_DEPENDS roscpp control_msgs cob_generic_can cob_canopen_motor cob_utilities libpcan std_msgs std_srvs sensor_msgs diagnostic_msgs message_runtime INCLUDE_DIRS common/include LIBRARIES ${PROJECT_NAME}
```
Comment (mgruhler) [inline cob_base_drive_chain/CMakeLists.txt]: Did roscompile put this into one line?
For the sake of readability, having a line for each of `CATKIN_DEPENDS`, `INCLUDE_DIRS`, `LIBRARIES` would be better, IMO.

File: cob_base_drive_chain/package.xml [hunk 2]
```diff
@@ -13,33 +13,32 @@
...
   <run_depend>cob_generic_can</run_depend>
   <run_depend>cob_canopen_motor</run_depend>
-  <run_depend>cob_utilities</run_depend>
-  <run_depend>message_runtime</run_depend>  
   <run_depend>libpcan</run_depend>
   <run_depend>libntcan</run_depend>
   
   <export>
-    <cpp lflags="-Wl,-rpath,${prefix}/common/lib -L${prefix}/common/lib -lcob_base_drive_chain" cflags="-I${prefix}/common/include"/>
+    <cpp cflags="-I${prefix}/common/include" lflags="-Wl,-rpath,${prefix}/common/lib -L${prefix}/common/lib -lcob_base_drive_chain"/>
```
Comment (mgruhler) [inline cob_base_drive_chain/package.xml]: Do we need those flags at anymore? I guess this is a leftover from `rosbuild`...

File: cob_canopen_motor/package.xml [hunk 3]
```diff
@@ -21,6 +21,6 @@
   <run_depend>cob_utilities</run_depend>
 
   <export>
-    <cpp lflags="-Wl,-rpath,${prefix}/common/lib -L${prefix}/common/lib -lcob_canopen_motor_harmonica" cflags="-I${prefix}/common/include"/>
+    <cpp cflags="-I${prefix}/common/include" lflags="-Wl,-rpath,${prefix}/common/lib -L${prefix}/common/lib -lcob_canopen_motor_harmonica"/>
```
Comment (mgruhler) [inline cob_canopen_motor/package.xml]: Do we need those flags at anymore? I guess this is a leftover from `rosbuild`...

File: cob_generic_can/package.xml [hunk 4]
```diff
@@ -22,8 +22,7 @@
   <run_depend>libpcan</run_depend>
   <run_depend>libntcan</run_depend>
 
-
   <export>
-    <cpp lflags="-Wl,-rpath,${prefix}/common/lib -L${prefix}/common/lib -lcob_generic_can_peaksysusb -lcob_generic_can_peaksys -lcob_generic_can_esd" cflags="-I${prefix}/common/include"/>
+    <cpp cflags="-I${prefix}/common/include" lflags="-Wl,-rpath,${prefix}/common/lib -L${prefix}/common/lib -lcob_generic_can_peaksysusb -lcob_generic_can_peaksys -lcob_generic_can_esd"/>
```
Comment (mgruhler) [inline cob_generic_can/package.xml]: Do we need those flags at anymore? I guess this is a leftover from `rosbuild`...

File: cob_relayboard/package.xml [hunk 5]
```diff
@@ -10,22 +10,17 @@
...
-  <run_depend>rostest</run_depend>
+  <run_depend>rospy</run_depend>
   <run_depend>std_msgs</run_depend>
-  <run_depend>cob_utilities</run_depend>
   <run_depend>cob_msgs</run_depend>
 
-
   <export>
-    <cpp lflags="-Wl,-rpath,${prefix}/common/lib -L${prefix}/common/lib -lSerialIO -lcob_relayboard" cflags="-I${prefix}/common/include"/>
+    <cpp cflags="-I${prefix}/common/include" lflags="-Wl,-rpath,${prefix}/common/lib -L${prefix}/common/lib -lSerialIO -lcob_relayboard"/>
```
Comment (mgruhler) [inline cob_relayboard/package.xml]: Do we need those flags at anymore? I guess this is a leftover from `rosbuild`...

File: cob_undercarriage_ctrl/CMakeLists.txt [hunk 6]
```diff
@@ -1,45 +1,23 @@
...
-## catkin specific configuration ##
-###################################
-
 catkin_package(
-    CATKIN_DEPENDS roscpp nav_msgs geometry_msgs diagnostic_msgs tf cob_msgs cob_utilities diagnostic_updater
-    INCLUDE_DIRS # TODO include
-    LIBRARIES # TODO
+  CATKIN_DEPENDS roscpp nav_msgs geometry_msgs diagnostic_msgs tf cob_msgs cob_utilities diagnostic_updater control_msgs INCLUDE_DIRS # TODO include
+  LIBRARIES ${PROJECT_NAME}
+  INCLUDE_DIRS common/include
```
Comment (mgruhler) [inline cob_undercarriage_ctrl/CMakeLists.txt:9]: Here are now two `INCLUDE_DIRS` tags

File: cob_undercarriage_ctrl/package.xml [hunk 7]
```diff
@@ -26,14 +26,14 @@
...
   <run_depend>diagnostic_msgs</run_depend>
   <run_depend>tf</run_depend>
   <run_depend>cob_msgs</run_depend>
   <run_depend>cob_utilities</run_depend>
   <run_depend>diagnostic_updater</run_depend>
 
-
   <export>
-    <cpp lflags="-Wl,-rpath,${prefix}/common/lib -L${prefix}/common/lib -lcob_undercarriage_ctrl" cflags="-I${prefix}/common/include"/>
+    <cpp cflags="-I${prefix}/common/include" lflags="-Wl,-rpath,${prefix}/common/lib -L${prefix}/common/lib -lcob_undercarriage_ctrl"/>
```
Comment (mgruhler) [inline cob_undercarriage_ctrl/package.xml]: Do we need those flags at anymore? I guess this is a leftover from `rosbuild`...

File: cob_utilities/package.xml [hunk 8]
```diff
@@ -12,12 +12,11 @@
...
   <author>Christian Connette</author>
 
-
   <buildtool_depend>catkin</buildtool_depend>
   <build_depend>roscpp</build_depend>
   <run_depend>roscpp</run_depend>
 
   <export>
-    <cpp lflags="-Wl,-rpath,${prefix}/common/lib -L${prefix}/common/lib -lcob_utilities" cflags="-I${prefix}/common/include"/>
+    <cpp cflags="-I${prefix}/common/include" lflags="-Wl,-rpath,${prefix}/common/lib -L${prefix}/common/lib -lcob_utilities"/>
```
Comment (mgruhler) [inline cob_utilities/package.xml]: Do we need those flags at anymore? I guess this is a leftover from `rosbuild`...

File: cob_base_drive_chain/package.xml (hunk 2 above)
Comment (fmessmer) [inline cob_base_drive_chain/package.xml]: they can be removed

File: cob_undercarriage_ctrl/CMakeLists.txt (hunk 6 above)
Comment (fmessmer) [inline cob_undercarriage_ctrl/CMakeLists.txt:9]: fixed

**Comment by floweisshardt** [comment]
hard to oversee if anything is missing. I'll merge and give it a try on the buildfarm.

**Comment by NadiaHG** [comment]
I added it again , also in CMakeLists and package.xml, is it right?
https://github.com/ipa320/cob_driver/pull/233

**Pull request merged**